financeextras.php

```diff
@@ -273,6 +273,20 @@ function financeextras_civicrm_alterMailContent(&$content) {
   if (($content['workflow_name'] ?? NULL) === 'contribution_offline_receipt') {
     $content['html'] = str_replace('$formValues.total_amount', '$contribution.total_amount', $content['html']);
   }
+
+  if (($content['workflow_name'] ?? NULL) === 'contribution_invoice_receipt') {
+    $path = E::path('/templates/CRM/Financeextras/MessageTemplate/SalesTaxConversionRateTable.tpl');
+    $content['html'] = str_replace('{financeextras.tax_exchange_rate_table}', file_get_contents($path), $content['html']);
+  }
+}
+
+/**
+ * Implements hook_civicrm_tokens().
+ */
+function financeextras_civicrm_tokens(&$tokens) {
```

Comment:
This hook is deprecated : https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_tokens/
 
you need to use `civi.token.list` and  `civi.token.eval` events instead: https://docs.civicrm.org/dev/en/latest/framework/token/#defining-tokens

Reply:
Great